Source/bmalloc/bmalloc/bmalloc.h

```diff
@@ -75,9 +83,16 @@ BINLINE void* tryMalloc(size_t size, CompactAllocationMode mode, HeapKind kind =
 BINLINE void* malloc(size_t size, CompactAllocationMode mode, HeapKind kind = HeapKind::Primary)
 {
 #if BUSE(LIBPAS)
-    if (!isGigacage(kind))
-        return bmalloc_allocate_inline(size, asPasAllocationMode(mode));
-    return bmalloc_allocate_auxiliary_inline(&heapForKind(gigacageKind(kind)), size, asPasAllocationMode(mode));
+    if (!isGigacage(kind)) {
+        void* allocation = bmalloc_allocate_inline(size, asPasAllocationMode(mode));
+        if (mode == CompactAllocationMode::NonCompact && MAR_isAddressInQualifyingPage(allocation))
```

Comment:
In pursuance of keeping this function a leaf node, perhaps this check could be folded into a function a la pas_zero which would wrap the underlying allocator function and itself have a 'casual case' slow path?